packages/mobile/src/store/offline-downloads/sagas/removeAllDownloadedFavoritesSaga.ts

```diff
@@ -0,0 +1,127 @@
+import type { ID } from '@audius/common'
+import { takeEvery, select, put } from 'typed-redux-saga'
+
+import {
+  getOfflineCollections,
+  getOfflineFavoritedCollections,
+  getOfflineTracks
+} from '../selectors'
+import type {
+  CollectionId,
+  CollectionReasonsToUpdate,
+  TrackReasonsToUpdate
+} from '../slice'
+import {
+  OfflineDownloadStatus,
+  updateCollectionDownloadReasons,
+  removeCollectionDownloads,
+  updateTrackDownloadReasons,
+  removeTrackDownloads,
+  removeAllDownloadedFavorites
+} from '../slice'
+
+/*
+ * Saga initiated when user has requested to un-download their favorites.
+ * This includes the "favorites" collection and all collections that were
+ * not explicitly downloaded by the user previously.
+ */
+export function* watchRemoveAllDownloadedFavorites() {
+  yield* takeEvery(
+    removeAllDownloadedFavorites.type,
+    removeAllDownloadedFavoritesWorker
+  )
+}
+
+function* removeAllDownloadedFavoritesWorker() {
+  yield* removeAllFavoritesTracks()
+  yield* removeAllFavoritesCollections()
+}
+
+function* removeAllFavoritesTracks() {
+  const offlineTracks = yield* select(getOfflineTracks)
+  const offlineTrackList = Object.keys(offlineTracks).map(
+    (offlineTrackId) => offlineTracks[offlineTrackId]
+  )
+
+  const tracksToRemove: ID[] = []
+  const reasonsToUpdate: TrackReasonsToUpdate[] = []
+
+  // TODO: double check if "favorites" collection is in here
+  const offlineFavoritedCollections = yield* select(
+    getOfflineFavoritedCollections
+  )
+  const favoritedCollectionIds = new Set([
+    'favorites',
+    ...Object.keys(offlineFavoritedCollections)
+  ])
+
+  offlineTrackList.forEach((offlineTrack) => {
```

Comment:
Apparently `for ... of` is more performant than `.forEach`

Reply:
down with that